data/mods/MindOverMatter/effectoncondition/eoc_mental_engineering.json

```diff
@@ -0,0 +1,5106 @@
+[
+  {
+    "type": "effect_on_condition",
+    "id": "EOC_TELEPATH_MENTAL_ENGINEERING_MESSAGE",
+    "effect": [ { "u_message": "You begin rewiring your brain to change your way of looking at the world.", "type": "neutral" } ]
+  },
+  {
+    "type": "effect_on_condition",
+    "id": "EOC_TELEPATH_MENTAL_ENGINEERING_SUCCESS_MESSAGE",
+    "effect": [ { "u_message": "You now see the world in an entirely different way.", "type": "neutral" } ]
+  },
+  {
+    "type": "effect_on_condition",
+    "id": "EOC_TELEPATH_MENTAL_ENGINEERING_SIDE_EFFECTS",
+    "condition": { "x_in_y_chance": { "x": { "math": [ "u_spell_level_sum('school': 'TELEPATH')" ] }, "y": 200 } },
```

Comment:
Unless I'm misreading this, this will make side effects _more_ likely the more powerful you are, eventually hitting 200 out of 200 and always triggering as your telepathic power grows. It should probably be something like `{ "x": { "math": [ "max( ( 200 - u_spell_level_sum('school': 'TELEPATH') ), 20)" ] } }`

In addition, the actual chance of side effects is much much lower than 50%: since the number picked is random, the odds that the character qualifies for the particular side effect that was picked are pretty low (Like, if you roll case 4 to lose Killer Drive and don't have Killer Drive, you've gotten off with no consequences). If a side effect is rolled, you'll need a way to reroll the list until an appropriate side effect that actually applies is picked.

Reply:
Fixed the math on the consequences EOC.

In terms of dealing with side effects not being given, I've made some improvements to it.  First, I got rid of most of the positive options leaving only negative and roleplay negative (cannibal may mechanically wise be a good trait but most real people really wouldn't want to mind control themselves into one) options.  Second, I use u_mutate_towards for each selector, which should ensure that if it is picked then the only way the player doesn't have some mutation towards it is if they already have it.  Likewise, there's no more 'remove trait' options which should dramatically reduce the likelihood of an option not doing anything.

There's still some chances of a choice not doing anything specifically because the play already has the rolled negative trait, but I think its incredibly less likely.  Since there's now 37 options each obtained negative trait increases the chance of failure by ~2%.  There's probably a way to make that 0%, but I also think that a character who is already a psycopathic cannibal who's afraid of the dark and a very slow learner has less chances to screw up their head because there's already so many things broken that there's less things left to screw up.